python/fakejuju/fixture.py

```diff
@@ -35,31 +47,87 @@ class JujuMongoDB(MongoDB):
     """
 
     def __init__(self):
+        serverPem = os.path.join(_get_cert_dir(), "server.pem")
+        args = JUJU_MONGOD_ARGS + (
+            "--sslPEMKeyFile={serverPem}".format(serverPem=serverPem),)
         super(JujuMongoDB, self).__init__(
             # We honor the JUJU_MONGOD environment variable, the same one
             # supported by github.com/juju/testing/mgo.go, and set by the
             # fake-juju Travis job (see .travis.yml).
             mongod=os.environ.get("JUJU_MONGOD", JUJU_MONGOD),
-            args=JUJU_MONGOD_ARGS,
+            args=args,
         )
         self.setClientKwargs(ssl=True, ssl_cert_reqs=ssl.CERT_NONE)
 
 
 class FakeJuju(Service):
     """Spawn a fake-juju process, pointing to the given mongodb port."""
 
-    def __init__(self, mongo_port, binary=FAKE_JUJUD, port=17099, **kwargs):
+    def __init__(self, version="2.0.2", mongo=None, port=17100, **kwargs):
         """
-        :param mongo_port: Port of the MongoDB instance that this fake-jujud
-            process should use.
-        :param binary: Path to the fake-jujud binary to spawn.
-        :param port: Port that the fake juju API server will listen to.
+        :param version: The version number of the specific fake-juju to use.
+        :param mongo: The JujuMongoDB fixture instance managing the mongod
+            process to point this fake-juju to. If None, a brand new one
+            will be created.
+        :param port: Port that the fake juju control plane API will listen to.
```

Comment:
Fixed (note that mongo will use a random port, the juju API api server will use port - 1)